napari_animation/utils.py

```diff
@@ -67,3 +69,82 @@ def keys_to_list(input_dict):
                 yield [key] + sub_key
         else:
             yield [key]
+
+
+def quaternion2euler(quaternion, degrees=False):
+    """Converts scipy quaternion into euler angle representation.
+
+    Euler angles have degeneracies, so the output might different
+    from the Euler angles that might have been used to generate
+    the input quaternion.
+
+    Euler angles representation also has a singularity
+    near pitch = Pi/2 ; to avoid this, we set to Pi/2 pitch angles
+     that are closer than the chosen epsilon from it.
+
+    Parameters
+    ----------
+    quaternion : list
+        Quaternion for conversion in order [qx, qy, qz, qw].
+    degrees : bool
+        If output is returned in degrees or radians.
+
+    Returns
+    -------
+    angles : 3-tuple
+        Euler angles in (rx, ry, rz) order.
+    """
+    epsilon = 1e-10
+
+    q = quaternion
+
+    sin_theta_2 = 2 * (q[3] * q[1] - q[2] * q[0])
+    sin_theta_2 = np.sign(sin_theta_2) * min(abs(sin_theta_2), 1)
+
+    if abs(sin_theta_2) > 1 - epsilon:
+        theta_1 = -np.sign(sin_theta_2) * 2 * np.arctan2(q[0], q[3])
+        theta_2 = np.arcsin(sin_theta_2)
+        theta_3 = 0
+
+    else:
+        theta_1 = np.arctan2(
+            2 * (q[3] * q[2] + q[1] * q[0]),
+            1 - 2 * (q[1] * q[1] + q[2] * q[2]),
+        )
+
+        theta_2 = np.arcsin(sin_theta_2)
+
+        theta_3 = np.arctan2(
+            2 * (q[3] * q[0] + q[1] * q[2]),
+            1 - 2 * (q[0] * q[0] + q[1] * q[1]),
+        )
+
+    angles = (theta_1, theta_2, theta_3)
+
+    if degrees:
+        return tuple(np.degrees(angles))
+    else:
+        return angles
```

Comment:
looks like this came in from a merge - have removed in tlambert-forks/napari-animation#1